PIPS-NLP/Core/NlpStoch/sLinsysRootAugSparseRowMaj.C

```diff
@@ -455,60 +455,60 @@ void sLinsysRootAugSpTriplet::reduceKKT()
     }
     //printf("Rank %d  has %d entries that rank0 does not have.\n", myRank, nnz_diff);
   }
-  // - the common entries are MPI_Reduce-d
-  {
-    double* doublebuffer = NULL;
-    if(iAmRank0) {
-      doublebuffer = new double[nnzRoot];
-    }
-    MPI_Reduce(M, doublebuffer, nnzRoot, MPI_DOUBLE, MPI_SUM, 0, mpiComm);
-    if(doublebuffer) memcpy(M, doublebuffer, nnzRoot*sizeof(double));
-    delete [] doublebuffer;
-
-    if(iAmRank0) {
-      bool bret = kktm.atPutSparseTriplet(irn,jcn,M,nnzRoot);
-      assert(bret==true && "sparsity patterns do not match on rank 0; this should not happen");
-    }
-  }
 
   // - the saved entries are then MPI_Gather-ed 
   {
     int mismatch=(nnz_diff>0), auxG;
-    MPI_Allreduce(&mismatch, &auxG, 1, MPI_INT, MPI_MAX, mpiComm);
+    MPI_Allreduce(&mismatch, &auxG, 1, MPI_INT, MPI_MAX, mumpsComm);
     mismatch=auxG;
     if(mismatch) {
 
       //first get the sum of nnz of diff over all processes -> needed to allocate recv buffer at root
-      int commSize; MPI_Comm_size( mpiComm, &commSize); 
+      int commSize; MPI_Comm_size( mumpsComm, &commSize); 
       int* diff_counts = NULL, *irow_diff_dest=NULL, *jcol_diff_dest=NULL; double* M_diff_dest=NULL;
       if(iAmRank0) diff_counts = new int[commSize];
 
-      MPI_Gather(&nnz_diff, 1, MPI_INT, diff_counts, 1, MPI_INT, 0, mpiComm);
+      MPI_Gather(&nnz_diff, 1, MPI_INT, diff_counts, 1, MPI_INT, 0, mumpsComm);
 
-      int* displs=NULL;
+      size_t* displs=NULL;
       if(iAmRank0) {
 	assert(diff_counts[0]==0); //diff for rank 0 should be empty
-	displs = new int[commSize];
+	displs = new size_t[commSize];
 
 	displs[0]=0;
 	for(int i=1; i<commSize; i++) { 
 	  displs[i] = displs[i-1] + diff_counts[i-1];
-	  //printf("Rank %d  -> diff_counts=%d\n", i, diff_counts[i]);
+	  printf("Rank %d  -> displs=%zu , diff_counts=%d\n", i, displs[i], diff_counts[i]);
 	}
 
-	int nnz_diff_total = displs[commSize-1]+diff_counts[commSize-1];
+	size_t nnz_diff_total = displs[commSize-1]+diff_counts[commSize-1];
 
 	//printf("Rank %d  -> %d entries to be gathered\n", myRank, nnz_diff_total);
 
 	irow_diff_dest = new int[nnz_diff_total];
 	jcol_diff_dest = new int[nnz_diff_total];
+  // Only used with all set to zero. Handed over to atAddSpRow. One could avoid this.
```

Comment:
indentation off?

Reply:
Once I started, I just fixed the indentation everywhere. I hope that's okay.